app/ante/evm/04_validate.go

```diff
@@ -99,13 +98,19 @@ func ValidateTx(tx sdktypes.Tx) (*tx.Fee, error) {
 	return authInfo.Fee, nil
 }
 
+// CheckTxFee checks if the Amount and GasLimit fields of the txFeeInfo input
+// are equal to the txFee coins and the txGasLimit value.
+// The function expects txFeeInfo to contains coins in the original decimal
+// representation.
 func CheckTxFee(txFeeInfo *tx.Fee, txFee sdktypes.Coins, txGasLimit uint64) error {
 	if txFeeInfo == nil {
 		return nil
 	}
 
-	if !txFeeInfo.Amount.Equal(txFee) {
-		return errorsmod.Wrapf(errortypes.ErrInvalidRequest, "invalid AuthInfo Fee Amount (%s != %s)", txFeeInfo.Amount, txFee)
+	convertedAmount := evmtypes.ConvertCoinsTo18Decimals(txFeeInfo.Amount)
+
+	if !convertedAmount.Equal(txFee) {
```

Comment:
This should be the other way around. We need to remove the extra precision of the txFee.
If we scale the txFeeInfo, we'll add some trailing zeros. And if we have non-zero decimals after the 6th decimal in the txFee, the equality will never be true, even though these are discarded anyways

Reply:
Good point @GAtom22, will update this!